cumulus/pallets/parachain-system/src/lib.rs

```diff
@@ -1648,6 +1648,23 @@ impl<T: Config> UpwardMessageSender for Pallet<T> {
 	fn send_upward_message(message: UpwardMessage) -> Result<(u32, XcmHash), MessageSendError> {
 		Self::send_upward_message(message)
 	}
+
+	fn can_send_upward_message(message: &UpwardMessage) -> Result<(), MessageSendError> {
+		#[cfg(not(feature = "runtime-benchmarks"))]
+		let max_upward_message_size = HostConfiguration::<T>::get()
+			.map(|cfg| cfg.max_upward_message_size)
+			.ok_or(MessageSendError::Other)?;
+		// for benchmarks, if host cfg is not set, we assume no limit on the size.
```

Comment:
@kianenigma this is not enough, because several tests fail now,
probably some more tuning needed for feature flags here, e.g.:
```
#[cfg(any(feature = "std", feature = "runtime-benchmarks", test))]

and also cover `not(any)`
```

Btw. I fixed tests also here: https://github.com/paritytech/polkadot-sdk/pull/8458 with approach 2